### DISCUSSION TITLE
error "has dtype 'category' but is not in cat_features list"

### DISCUSSION THREAD

**Discussion by tom-dwyer** [catboost/catboost#2788] [General]
I kept getting the error "has dtype 'category' but is not in cat_features list" and realized the catboost_model.get_cat_feature_indices() was off by one.  The target variable was the first column in the table.  When you move that column to the end of the table (or beyond the last categorical variable), the index is fine.  It's a weird fix, but it works!!!!

**Comment by andrey-khropov** [comment]
When `cat_features`, `text_features` and `embedding_features` parameters contain indices these are feature indices.

What to you mean by `table` in this context?

**Reply by tom-dwyer** [reply]
Sorry I meant in the data frame.  So when the target variable is the first column in the dataframe, and say i have 3 categorical features in the next 3 columns followed by 5 numeric features, the model.cat_features would be [0,1,2].  When trying to use  PartialDependenceDisplay.from_estimator(catboost_model, train_data, [feature]) this would cause it to crash with the error "has dtype 'category' but is not in cat_features list" .  If I move the target variable to the last column, the model.cat_features would STILL be [0,1,2] and the partial dependency plots work fine.  The suggests that the cat_features needs to be adjusted, but can work if the target variable is simply moved to the last column
